test/UnitTests/GitHub.TeamFoundation/VSGitExtTests.cs

```diff
@@ -0,0 +1,151 @@
+            var sp = CreateGitHubServiceProvider(context);
+
+            var target = new VSGitExt(sp);
+            var eventArgs = Substitute.For<IVSUIContextChangedEventArgs>();
+            eventArgs.Activated.Returns(activated);
+            context.UIContextChanged += Raise.Event<EventHandler<IVSUIContextChangedEventArgs>>(context, eventArgs);
+
+            sp.Received(expectCalls).GetService<IGitExt>();
+        }
+    }
+
+    public class TheActiveRepositoriesChangedEvent
+    {
+        [Test]
+        public void GitExtPropertyChangedEvent_ActiveRepositoriesChangedIsFired()
+        {
+            var context = CreateVSUIContext(true);
+            var gitExt = CreateGitExt();
+            var sp = CreateGitHubServiceProvider(context, gitExt);
+            var target = new VSGitExt(sp);
+
+            bool wasFired = false;
+            target.ActiveRepositoriesChanged += () => wasFired = true;
+            var eventArgs = new PropertyChangedEventArgs(nameof(gitExt.ActiveRepositories));
+            gitExt.PropertyChanged += Raise.Event<PropertyChangedEventHandler>(gitExt, eventArgs);
+
+            Assert.That(wasFired, Is.True);
+        }
+
+        [Test]
+        public void WhenUIContextChanged_ActiveRepositoriesChangedIsFired()
+        {
+            var context = CreateVSUIContext(false);
+            var gitExt = CreateGitExt();
+            var sp = CreateGitHubServiceProvider(context, gitExt);
+            var target = new VSGitExt(sp);
+
+            bool wasFired = false;
+            target.ActiveRepositoriesChanged += () => wasFired = true;
+
+            var eventArgs = Substitute.For<IVSUIContextChangedEventArgs>();
+            eventArgs.Activated.Returns(true);
+            context.UIContextChanged += Raise.Event<EventHandler<IVSUIContextChangedEventArgs>>(context, eventArgs);
+
+            Assert.That(wasFired, Is.True);
+        }
+    }
+
+    public class TheActiveRepositoriesProperty
+    {
+        [Test]
+        public void SccProviderContextNotActive_IsNull()
+        {
+            var context = CreateVSUIContext(false);
+            var sp = CreateGitHubServiceProvider(context);
+
+            var target = new VSGitExt(sp);
+
+            Assert.That(target.ActiveRepositories, Is.Null);
+        }
+
+        [Test]
+        public void SccProviderContextIsActive_DefaultEmptyListFromGitExt()
+        {
+            var context = CreateVSUIContext(true);
+            var gitExt = CreateGitExt(new string[0]);
+            var sp = CreateGitHubServiceProvider(context, gitExt);
+            var target = new VSGitExt(sp);
+
+            var activeRepositories = target.ActiveRepositories;
+
+            Assert.That(activeRepositories.Count(), Is.EqualTo(0));
+        }
+
+        // TODO: We can't currently test returning a non-empty list because it constructs a live LocalRepositoryModel object.
+        // Move could move the responsibility for constructing a LocalRepositoryModel object outside of VSGitExt.
```

Comment:
"Move could move"?

Reply:
Given that `VSGitExt` is simply a wrapper for `IGitExt` it might be worth exposing the `IGitRepositoryInfo`s directly instead of wrapping them in `ILocalRepositoryModel`s?

Reply:
I don't think we should be `ILocalRepositoryModels` here, but the problem with `IGitRepositoryInfo` is there's two versions. Maybe create another wrapper.